src/emmy/expression/compile.cljc

```diff
@@ -223,25 +234,147 @@
 ;; I.E., first the structure, then a vector of the original function's arguments.
 
 (defn- state-argv
-  "Returns the argument vector for a compiled state function, given:
-
-  - `params`: a seq of symbols equal in count to the original state function's
-    args
-  - `state-model`: a sequence of variables representing the structure of the
-    nested function returned by the state function
-  - `opts`, a dictionary of compilation options.
-
-  See [[compile-state-fn*]] for a description of the options accepted in
-  `opts`."
-  [params state-model {:keys [flatten? generic-params?]
-                       :or {flatten? true
-                            generic-params? true}}]
-  (let [state (into [] (if flatten?
-                         (flatten state-model)
-                         state-model))]
-    (if generic-params?
-      [state (into [] params)]
-      [state])))
+  "Computes the argument list corresponding to `state-model` according to
+  the `:calling-convention` in the code object. The argument list will be
+  placed stored in the `:argv` key, and the original state model itself
+  in `:state-model`.
+
+  - `state-model`: a structure of symbols representing the state value
+    operated on by the state derivative
+
+  Effectively, there are four classes of signature handled here.
+
+  In each of the following displays, the parameter argument list
+  `[p0...]` will be present only if the key `:params` is present
+  on the code object, corresponding to the use of the
+  `:generic-params?` compilation option.
+
+  - `:structure => (fn [[t [x0 x1] [v0 v1]] [p0 p1 ...]] ...) `
+
+  This is the typical signature for a state derivative discussed in
+  SICM. It maps a state tuple to its time derivative; structure in,
+  structure out.
+
+  - `:flat => (fn [[t x0 x1 v0 v1] [p0 p1 ...]] ...)`
+
+  ODE solvers prefer to work with flat vectors, and so this calling
+  convention unrolls the state model into an ordered list of parameters.
+
+  - `:primitive => (fn [ys yps ps])`
```

Comment:
what does yps stand for?

Reply:
"y primes" (maybe I should spell it `yprimes`